tests/kademlia.rs

```diff
@@ -60,102 +101,178 @@ async fn find_peer_dht() {
     // one another in a chain; node 0 is only aware of the existence
     // of node 1 and so on; bootstrap them eagerly/quickly, so that
     // they don't have a chance to form the full picture in the DHT
-    for i in 0..CHAIN_LEN {
-        let (next_id, next_addr) = if i < CHAIN_LEN - 1 {
+    for i in 0..count {
+        let (next_id, next_addr) = if i < count - 1 {
             ids_and_addrs[i + 1].clone()
         } else {
             // the last node in the chain also needs to know some address
             // in order to bootstrap, so give it its neighbour's information
             // and then bootstrap it as well
-            ids_and_addrs[CHAIN_LEN - 2].clone()
+            ids_and_addrs[count - 2].clone()
         };
 
         nodes[i].add_peer(next_id, next_addr).await.unwrap();
         nodes[i].bootstrap().await.unwrap();
     }
 
-    // node 0 now tries to find the address of the very last node in the
-    // chain; the chain should be long enough for it not to automatically
-    // be connected to it after the bootstrap
-    let found_addrs = nodes[0]
-        .find_peer(ids_and_addrs[CHAIN_LEN - 1].0.clone())
-        .await
-        .unwrap();
+    // make sure that the nodes are not actively connected to each other
+    // and that we are actually going to be testing the DHT here
+    for node in &nodes {
+        assert!([1usize, 2].contains(&node.peers().await.unwrap().len()));
+    }
 
-    assert_eq!(found_addrs, vec![ids_and_addrs[CHAIN_LEN - 1].1.clone()]);
+    (nodes, ids_and_addrs, None)
 }
 
-// TODO: split into separate, more advanced bootstrap and closest_peers
-#[tokio::test(max_threads = 1)]
-async fn kademlia_local_peer_discovery() {
-    const BOOTSTRAPPER_COUNT: usize = 20;
+#[cfg(feature = "test_dht_with_go")]
```

Comment:
There shouldn't be any issue running this test on CI apart from some scripting, did you consider that? I think we should have curl, tar and friends available on all three platforms.

Reply:
I did, just wanted to make this self-contained first and add a CI entry as a follow-up

Reply:
Good good